lib/ts/recipe/thirdparty/providers/apple.ts

```diff
@@ -37,45 +37,41 @@ type TypeThirdPartyProviderAppleConfig = {
 export default function Apple(config: TypeThirdPartyProviderAppleConfig): TypeProvider {
     const id = "apple";
 
-    function getClientSecret(clientId: string, keyId: string, teamId: string, privateKey: string): string {
-        return jwtSign(
-            {
-                iss: teamId,
-                iat: Math.floor(Date.now() / 1000),
-                exp: Math.floor(Date.now() / 1000) + 86400 * 180, // 6 months
-                aud: "https://appleid.apple.com",
-                sub: getActualClientIdFromDevelopmentClientId(clientId),
-            },
-            privateKey.replace(/\\n/g, "\n"),
-            { algorithm: "ES256", keyid: keyId }
-        );
-    }
-    try {
-        // trying to generate a client secret, in case client has not passed the values correctly
-        getClientSecret(
-            config.clientId,
-            config.clientSecret.keyId,
-            config.clientSecret.teamId,
-            config.clientSecret.privateKey
-        );
-    } catch (error) {
-        throw new STError({
-            type: STError.BAD_INPUT_ERROR,
-            message: error.message,
-        });
+    async function getClientSecret(
+        clientId: string,
+        keyId: string,
+        teamId: string,
+        privateKey: string
+    ): Promise<string> {
+        const alg = "ES256";
+
+        return new jose.SignJWT({})
+            .setProtectedHeader({ alg, kid: keyId, typ: "JWT" })
+            .setIssuer(teamId)
+            .setIssuedAt()
+            .setExpirationTime("180days")
+            .setAudience("https://appleid.apple.com")
+            .setSubject(getActualClientIdFromDevelopmentClientId(clientId))
+            .sign(crypto.createPrivateKey(privateKey.replace(/\\n/g, "\n")));
     }
 
-    function get(redirectURI: string | undefined, authCodeFromRequest: string | undefined): TypeProviderGetResponse {
+    // trying to generate a client secret, in case client has not passed the values correctly
+    const clientSecretPromise = getClientSecret(
```

Comment:
in the older impl, if the config related to the secret provided by the user was invalid, then calling Apple.init({...}) itself would fail - does this still happen? (given that this is an async function now. If it doesn't happen anymore, that's OK - will the error now be thrown when we actually try and call the `await get` function?

Reply:
yeah, it throws on the first get, which is at the start of the first sign-in process using the provider.